### PR TITLE
Fixes paths names

### PR DESCRIPTION
**Pull request by avtar**
These changes update configuration file names and paths based on feedback from @cindyli

### CODE REVIEW

**Comment by avtar** [comment]
@amatas, @gtirloni, @mrtyler: if any of you have time to take a look at this that would be appreciated. @cindyli tested these changes.

File: tasks/deploy.yml [hunk 1]
```diff
@@ -45,14 +45,14 @@
 
 - name: Update the Preferences Server base config file's authDBServerPort
   replace:
-    dest: "{{ nodejs_app_install_dir }}/{{ preferences_server_config_dir }}/gpii.preferencesServer.config.base.json"
+    dest: "{{ nodejs_app_install_dir }}/{{ preferences_server_config_dir }}/{{ preferences_server_environment }}.json"
```
Comment (mrtyler) [inline tasks/deploy.yml]: Looks like these should be single quotes for consistency with the rest of the file. :)

File: tasks/deploy.yml [hunk 2]
```diff
@@ -45,14 +45,14 @@
...
     regexp: '"authDBServerPort": 8058'
     replace: '"authDBServerPort": {{ preferences_server_couchdb_host_port }}'
   notify: Restart the application systemd unit
 
 - name: Update the Preferences Server base config file's preferencesServer.dataStore.baseUrl
   replace:
-    dest: "{{ nodejs_app_install_dir }}/{{ preferences_server_config_dir }}/gpii.preferencesServer.config.base.json"
+    dest: "{{ nodejs_app_install_dir }}/{{ preferences_server_config_dir }}/{{ preferences_server_environment }}.json"
```
Comment (mrtyler) [inline tasks/deploy.yml]: Ditto.

**Comment by gtirloni** [comment]
I'm really confused about GPII config files. I had to change everything from gpii/configs to node_modules/*/configs to make things work. Does this mean gpii/configs is indeed the right place for this?

**Review by mrtyler — COMMENTED** [review]
I'm not able to vouch for the correctness of these changes (e.g. I don't know if these new file names are right), but I can state that these changes look reasonable from an Ansible / logic perspective.

**Comment by gtirloni** [comment]
It looks like the config file in this PR is only adding the other config file that we are pointing to (following my last changes):

```
$ cat gpii/configs/gpii.config.preferencesServer.standalone.production.json 
{
    "type": "gpii.config.preferencesServer.standalone.production",
    "mergeConfigs": [
        "%preferencesServer/configs/gpii.preferencesServer.config.production.json"
    ]
```


@cindyli can you put me out of my misery and explain and/or point to relevant docs so I can understand the rationale for these files?

**Comment by amatas** [comment]
From the point of view of Ansible role, it looks good to me as it works.

But if we talked in the past about getting rid of making strings substitutions in config files, these changes add another config file that also needs the replacement of some strings.

I'm not able to see the advantages of these changes, sorry. I join to the @gtirloni 's request for explanation.

**Comment by cindyli** [comment]
@gtirloni and @amatas, from my understanding, the config files to start GPII in certain modes are all  in `gpii/configs` directory. They assemble the needed config files from individual `gpii/node_modules/[gpii-module]/configs` directories and re-config them to adapt to the needed modes. The config files sitting in each `gpii/node_modules/[gpii-module]/configs` are for starting that module only. `gpii/configs/gpii.config.preferencesServer.standalone.production.json` happens to only include the `%preferencesServer/configs/gpii.preferencesServer.config.production.json` because it starts the prefs server in standalone. I agree with you using either one of these config files doesn't make much difference in this case. However, it would be nice to use the one from `gpii/configs` that is a common ground agreed by the team for holding config files to start GPII. Team members, please disagree if I'm wrong. :)

Also, the most important change in this pull request is not the file path and name. It's the one in `tasks/deploy.yml` that's to replace the hard-coded file name `gpii.preferencesServer.config.base.json` with `{{ preferences_server_environment }}.json`. As you can see, [this line](https://github.com/gpii-ops/ansible-preferences-server/blob/master/tasks/deploy.yml#L9) copies the config file to the destination in its original name instead of a fixed name. This is what causes the start of GPII production VM to fail.

**Comment by gtirloni** [comment]
Thanks for the summary, Cindy! So it looks like there is [agreement](https://wiki.gpii.net/w/GPII_Services_Configuration) that we should always use `gpii/configs`.

File: tasks/deploy.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@
 
     - name: Update the Preferences Server config file's CouchDB address
       replace:
-        dest: /tmp/gpii.preferencesServer.config.production.json
+        dest: '/tmp/{{ preferences_server_environment }}.json'
```
Comment (gtirloni) [inline tasks/deploy.yml:13]: Got this error starting the Docker container built with this change:

```
TASK [preferences-server : Update the Preferences Server production config file's CouchDB address] ***
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "Destination /opt/gpii/node_modules/universal/gpii/configs/gpii.preferencesServer.config.production.json does not exist !", "rc": 257}
```

**Comment by cindyli** [comment]
README needs to be modified to reflect the changes. It's still using old default values.

**Comment by gtirloni** [comment]
@cindyli one last question, I noticed that, although the config file is the same, the method of starting the application is different. And starting it through `node gpii.js`, registers more modules (see bellow). If we are interested in just spinning up a minimal/essential container running the Preferences Server alone, are these additional modules a problem?

The memory utilization at startup time goes from 62MB to 72MB (16%), which is not a lot but I don't know if this will spin up unnecessary functionality we don't want. 

# Using gpii/configs

Start: `node gpii.js`
Output: 

```
18:54:20.988:  Registering module universal from path /app
18:54:20.989:  Registering module infusion from path /app/node_modules/infusion
18:54:20.989:  Registering module infusion from path /app/node_modules/infusion
18:54:20.989:  Registering module universal from path /app
18:54:20.992:  Registering module deviceReporter from path /app/gpii/node_modules/deviceReporter
18:54:20.996:  Registering module kettle from path /app/node_modules/kettle
18:54:21.161:  Registering module lifecycleManager from path /app/gpii/node_modules/lifecycleManager
18:54:21.172:  Registering module lifecycleActions from path /app/gpii/node_modules/lifecycleActions
18:54:21.177:  Registering module flowManager from path /app/gpii/node_modules/flowManager
18:54:21.201:  Registering module gpii-oauth2 from path /app/gpii/node_modules/gpii-oauth2
18:54:21.318:  Registering module journal from path /app/gpii/node_modules/journal
18:54:21.340:  Registering module singleInstance from path /app/gpii/node_modules/singleInstance
18:54:21.347:  Registering module settingsHandlers from path /app/gpii/node_modules/settingsHandlers
18:54:21.356:  Registering module preferencesServer from path /app/gpii/node_modules/preferencesServer
18:54:21.357:  Registering module ontologyHandler from path /app/gpii/node_modules/ontologyHandler
18:54:21.369:  Registering module rawPreferencesServer from path /app/gpii/node_modules/rawPreferencesServer
18:54:21.386:  Registering module flatMatchMaker from path /app/gpii/node_modules/flatMatchMaker
18:54:21.389:  Registering module canopyMatchMaker from path /app/gpii/node_modules/canopyMatchMaker
18:54:21.393:  Registering module contextManager from path /app/gpii/node_modules/contextManager
18:54:21.397:  Registering module pouchManager from path /app/gpii/node_modules/pouchManager
18:54:21.623:  Registering request handler {
    "route": "/preferences/:userToken",
    "method": "get",
    "type": "gpii.preferencesServer.get.handler"
}
18:54:21.624:  Registering request handler {
    "route": "/preferences",
    "method": "post",
    "type": "gpii.preferencesServer.post.handler"
}
18:54:21.624:  Registering request handler {
    "route": "/preferences/:userToken",
    "method": "put",
    "type": "gpii.preferencesServer.put.handler"
}
18:54:21.624:  Registering request handler {
    "route": "/add-preferences",
    "method": "post",
    "type": "gpii.preferencesServer.add.post.handler"
}
18:54:21.626:  Opening Kettle Server on port 8081
18:54:21.629:  Kettle Server uccyoqp-44 is listening on port 8081
```

# Using gpii/node_modules/preferencesServer/configs

Start command: `node node_modules/kettle/init.js gpii/node_modules/preferencesServer/configs`
Output:

```
Infusion at path /app/node_modules/infusion is at top level 
18:53:49.743:  Registering module universal from path /app
18:53:49.744:  Registering module infusion from path /app/node_modules/infusion
18:53:49.744:  Registering module infusion from path /app/node_modules/infusion
18:53:49.747:  Registering module kettle from path /app/node_modules/kettle
18:53:49.898:  Registering module preferencesServer from path /app/gpii/node_modules/preferencesServer
18:53:49.899:  Registering module ontologyHandler from path /app/gpii/node_modules/ontologyHandler
18:53:49.905:  Registering module gpii-oauth2 from path /app/gpii/node_modules/gpii-oauth2
18:53:50.259:  Registering request handler {
    "route": "/preferences/:userToken",
    "method": "get",
    "type": "gpii.preferencesServer.get.handler"
}
18:53:50.260:  Registering request handler {
    "route": "/preferences",
    "method": "post",
    "type": "gpii.preferencesServer.post.handler"
}
18:53:50.260:  Registering request handler {
    "route": "/preferences/:userToken",
    "method": "put",
    "type": "gpii.preferencesServer.put.handler"
}
18:53:50.260:  Registering request handler {
    "route": "/add-preferences",
    "method": "post",
    "type": "gpii.preferencesServer.add.post.handler"
}
18:53:50.262:  Opening Kettle Server on port 8081
18:53:50.268:  Kettle Server c8e9oh7e-43 is listening on port 8081
```

**Comment by cindyli** [comment]
You are right, @gtirloni, using `node gpii.js` to start GPII will register all gpii modules regardless of which config file is used. See [here](https://github.com/GPII/universal/blob/master/index.js#L23-L40). As indicated in the code comment, [FLUID-5521](https://issues.fluidproject.org/browse/FLUID-5521) was created for addressing it. These additional modules are loaded but not used. They are unnecessary but won't cause a problem.

File: tasks/deploy.yml (hunk 3 above)
Comment (avtar) [reply tasks/deploy.yml:13]: It looks like the ``NODE_ENV`` environment variable needs to be used to point to this new file but the real issue is most likely this old default:

https://github.com/gpii-ops/docker-preferences-server/blob/master/provisioning/run-vars.yml#L5

[Line 9](https://github.com/gpii-ops/docker-preferences-server/blob/master/provisioning/run-vars.yml#L9) might also need to be updated depending on what is the preferred way of starting the Preferences Server.

**Review by gtirloni — APPROVED** [review]
LTGM

**Pull request merged**